### PR TITLE
IPV6 support RtpsRelay and ICE

### PR DESCRIPTION
**Pull request by jrw972**

### CODE REVIEW

File: dds/DCPS/RTPS/ICE/EndpointManager.cpp [hunk 1]
```diff
@@ -181,15 +181,33 @@ void EndpointManager::change_password(bool password_only)
...
+    if (pos->is_ipv4_compat_ipv6()) {
+      continue;
+    }
+
+    // TODO(jrw972):  Include once supported in ACE.
+    // if (pos->is_sitelocal()) {
+    //   continue;
+    // }
+
+#if !IPV6_ONLY
```
Comment (iguessthislldo) [inline dds/DCPS/RTPS/ICE/EndpointManager.cpp:204]: Where is this macro coming from?

Comment (iguessthislldo) [reply dds/DCPS/RTPS/ICE/EndpointManager.cpp:204]: If it's specific to OpenDDS and is going to stay could you rename it `OPENDDS_IPV6_ONLY`?

**Comment by mitza-oci** [comment]
Codacy/cppcheck - nothing needs to be done for these "issues"

File: dds/DCPS/RTPS/Spdp.cpp [hunk 2]
```diff
@@ -1760,10 +1817,12 @@ Spdp::SpdpTransport::open()
     throw std::runtime_error("failed to register unicast input handler");
   }
 
-  if (reactor->register_handler(multicast_socket_.get_handle(),
+#ifdef ACE_HAS_IPV6
+  if (reactor->register_handler(unicast_ipv6_socket_.get_handle(),
                                 this, ACE_Event_Handler::READ_MASK) != 0) {
-    throw std::runtime_error("failed to register multicast input handler");
+    throw std::runtime_error("failed to register unicast input handler");
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/Spdp.cpp:1823]:
```suggestion
    throw std::runtime_error("failed to register unicast IPv6 input handler");
```

**Comment by jwillemsen** [comment]
Looks something broke on Windows, see https://buildlogs.remedy.nl/win_msvc15_opendds_debug/. Maybe this PR or #1636

**Pull request merged**